src/prettytable/prettytable.py

```diff
@@ -1511,6 +1511,27 @@ def clear(self) -> None:
     def copy(self):
         return copy.deepcopy(self)
 
+    def get_output(self, format: str = "text", **kwargs) -> str:
```

Comment:
Kind of a pity `get_string` is already used for the plaintext version :)

Please could you rename `format` to something that doesn't shadow the builtin?

Reply:
Thanks for the rename suggestion -- I sometimes forget the builtins, since most editors do NOT warn you about them.

I think it would work if I renamed `get_string()` to `get_text_string()` and then `get_output()` to `get_string()`. Should I try that change out?

Reply:
Based on what the function is doing, I'd suggest renaming it to `get_formatted_output`.

Reply:
Would `get_formatted_string` be the better name, since it is calling a `get_*string` method?

Reply:
sounds good to me

Reply:
Here's a diff (on top of my PR) where I rename `get_string` to `get_text_string` and `get_output` to `get_string`. IMO, this makes the function names most accurate and concise. However, I'm a bit reluctant to suggest routing all the current `get_string` calls through my new function.

```diff
--- a/src/prettytable/prettytable.py
+++ b/src/prettytable/prettytable.py
@@ -1511,7 +1511,7 @@ class PrettyTable:
     def copy(self):
         return copy.deepcopy(self)
 
-    def get_output(self, out_format: str = "text", **kwargs) -> str:
+    def get_string(self, out_format: str = "text", **kwargs) -> str:
         """Return string representation of specified format of table in current state.
 
         Arguments:
@@ -1519,7 +1519,7 @@ class PrettyTable:
         kwargs - passed through to function that performs formatting
         """
         if out_format == "text":
-            return self.get_string(**kwargs)
+            return self.get_text_string(**kwargs)
         if out_format == "html":
             return self.get_html_string(**kwargs)
         if out_format == "json":
@@ -1692,7 +1692,7 @@ class PrettyTable:
     # PLAIN TEXT STRING METHODS  #
     ##############################
 
-    def get_string(self, **kwargs) -> str:
+    def get_text_string(self, **kwargs) -> str:
         """Return string representation of table in current state.
 
         Arguments:
```

Reply:
The important thing would be to ensure `get_string()` remains fully backwards compatible.

Because the old `get_string()` only uses `**kwargs` and no positional arguments, we might be safe? We could make `out_format` a keyword only as well.

---

An alternative could be to avoid renaming or adding functions. Instead add `out_format` to the existing `get_string()`, and immediately check its value and potentially return one of the other (json, csv, etc) functions.

I'm less sure of this from a documentation point of view: `get_string()`'s docstring lists lots of args which don't apply to ther other functions.

Reply:
I did not initially make the change to rename `get_string` to `get_text_string` because I was not 100% sure that none of the libraries called in those functions do NOT have an argument that shadows whatever name I choose. I figure the unit tests are a good way of testing, but not a guarantee. 

I guess the safest way to do this is introduce the new function that calls the existing function. I have renamed `get_output()` to `get_formatted_string()` to be more accurate (as suggested by @adilhussain540).

Reply:
> I did not initially make the change to rename `get_string` to `get_text_string` because I was not 100% sure that none of the libraries called in those functions do NOT have an argument that shadows whatever name I choose. I figure the unit tests are a good way of testing, but not a guarantee.

I wouldn't worry too much about that, the existing `get_*string` have dozens of arguments. And there's only a couple of libraries to check (csv and json?). If we wanted to be really careful we could call it `prettytable_format` or something.

But adding a new function is fine too.